src/components/main-map.tsx

```diff
@@ -630,7 +705,7 @@ export function MainMap(): JSX.Element {
       clearSpiderifiedCluster(mapRef.current)
       map.triggerRepaint()
     },
-    [allPoints.features, checkDates, filters, toggleLines],
+    [allPoints.features, checkDates, filters],
```

Comment:
the linter correctly complains here about `toggleLines` missing in the deps array. i guess leaving it out "fixed" the perf issues but this should be revisited in a follow-up